specification/quantum/resource-manager/Microsoft.Quantum/preview/2019-11-04-preview/quantum.json

```diff
@@ -0,0 +1,910 @@
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The unique id for this workspace.",
+          "readOnly": true
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of this workspace.",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of this workspace.",
+          "readOnly": true
+        },
+        "location": {
+          "type": "string",
+          "description": "The location (region) of this workspace."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Gets or sets the properties. Define quantum space's specific properties.",
+          "$ref": "#/definitions/WorkspaceResourceProperties"
+        },
+        "tags": {
+          "type": "object",
+          "description": "The tags associated with this workspace."
+        },
+        "identity": {
+          "type": "object",
+          "description": "Managed Identity information",
+          "properties": {
+            "type": {
+              "type": "string",
+              "description": "Type of identity. Supported: SystemAssigned or None (case insensitive)"
+            },
+            "principalId": {
+              "type": "string",
+              "description": "objectId of the managed identity,"
+            },
+            "tenantId": {
+              "type": "string",
+              "description": "tenantId where the managed identity lives."
+            }
+          }
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "WorkspaceResourceProperties": {
+      "description": "Properties of a Workspace",
+      "properties": {
+        "providers": {
+          "type": "array",
+          "description": "List of Providers selected for this Workspace",
+          "items": {
+            "$ref": "#/definitions/Provider"
+          }
+        },
+        "usable": {
+          "type": "string",
+          "enum": [
+            "Yes",
+            "No",
+            "Partial"
+          ],
+          "x-ms-enum": {
+            "name": "UsableStatus",
+            "modelAsString": true
+          },
+          "description": "Whether the current workspace is ready to accept Jobs."
+        },
+        "provisioningState": {
+          "type": "string",
+          "enum": [
+            "Succeeded",
+            "ProviderLaunching",
+            "ProviderUpdating",
+            "ProviderDeleting",
+            "ProviderProvisioning",
+            "Deleted",
+            "Failed"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningStatus",
+            "modelAsString": true
+          },
+          "description": "Provisioning status field"
+        },
+        "resourceUsageId": {
+          "type": "string",
+          "description": "Unique id to track resource usage."
+        },
+        "storageAccount": {
```

Comment:
consider storageAccountId - also how is this value used?

Reply:
Yes, the convention for any resource id is <prefix>Id to make it obvious that the user is dealing with a fully qualified resource id.

---
In reply to: [546934759](https://github.com/Azure/azure-rest-api-specs/pull/11266#discussion_r546934759) [](ancestors = 546934759)

Reply:
This is a breaking change for existing customers. We'll take this at a new api version after the preview.

Reply:
You cannot make breaking changes even between GA'd api-versions.

Reply:
@JeffreyRichter We're going public preview, not GA yet. Also we don't have end customers directly calling APIs, but a number of clients - python, .NET, CLI and portal extension. Handling this via a new api-version will help us fix this with no downtime to customers.

Reply:
OK then, I understand